clar2wasm/tests/wasm-generation/functions.rs

```diff
@@ -0,0 +1,60 @@
+use clar2wasm::tools::crosscheck;
+use clarity::vm::Value;
+use proptest::proptest;
+use proptest::strategy::Strategy;
+
+use crate::{random_expressions, PropValue, TypePrinter};
+
+const DEFINE_PRIVATE_READONLY: [&str; 2] = ["define-private", "define-read-only"];
+
+proptest! {
+    #![proptest_config(super::runtime_config())]
+
+    #[test]
+    fn define_private_readonly(
+        (v, (expr, expected_val, is_response_intermediary)) in PropValue::any().prop_flat_map(|v| {
+            random_expressions(20,false).prop_map(move |tuple| (v.clone(), tuple))
+        })
+) {
+        for function in DEFINE_PRIVATE_READONLY{
+            let expr=format!("(begin {expr})");
+            let snippet=&format!(r#"({function} (func (a {})) {expr}) (func {v})"#,v.type_string());
```

Comment:
You did the opposite of what should have been tested for a function declaration: you should have generated any number of argument and a simple expression for the body. Your goal is to test function definitions, not `begin`.

Reply:
Do you suggest that I should Ignore begin? I mean we should test intermediary expressions though. I'll take care of arguments

Reply:
You don't test intermediary expressions: `begin` is an expression. You test a single expression which is a `begin`.